mem/mem_linux.go

```diff
@@ -156,3 +159,39 @@ func SwapMemoryWithContext(ctx context.Context) (*SwapMemoryStat, error) {
 	}
 	return ret, nil
 }
+
+func calcuateAvailVmem(ret *VirtualMemoryStat) uint64 {
+	var watermarkLow uint64
+	fn := common.HostProc("zoneinfo")
+	lines, _ := common.ReadLines(fn)
```

Comment:
from psutil, [kernel < 2.6.13 zoneinfo is not included](https://github.com/giampaolo/psutil/pull/890/files#diff-befef210ffc1d19a84ee5969886021a3R322). So could you add err check and if err, fall back to original `ret.Free + ret.Buffers + ret.Cached` ?

Reply:
@shirou such a good point. `calcuateAvailVmem` should check kernel version and error. The fallback will be needed if some error or under kernel version 2.6.13. 

But, `psutil` fallback is the sum of `free` and `cached`, not `buffer`. I think that `buffer` is also needed to be add to fallback. Adding `buffer` is more precise calculation, but they didn't. I need to know why they didn't.

Reply:
hmm. So something changed after 2.6.13 about calculation, perhaps. Then, could you change to the sum of free and cached, same as psutil?

Reply:
@shirou I found out that [very old linux kernel calculates memory available adding free and cached](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=34e431b0ae398fc54ea69ff85ec700722c9da773). I should have read the document carefully. 

Adding free and cached is sufficient calculating memory available for old kernel and that is more suitable for kernel. 

I will commit your review soon.

Reply:
@shirou Sorry for late commit. Previous my commit wrongly implemented the algorithm. [I should have used `active(file)` and `inactive(file)` metrics instead of active, inactive](https://github.com/giampaolo/psutil/blob/master/psutil/_pslinux.py#L352). 
[On kernel 2.6.28, the kernel specified active and inactive metrics to active(file), active(anon), inactive(file), and inactive(anon)](https://kernelnewbies.org/Linux_2_6_28#Memory_management_Scalability_improvements), I overlooked that. 
`To properly calculate available memory, active(file) and inactive(file) are needed and these should be added to Memory struct in mem.go